addons/mail/static/src/core/common/thread.js

```diff
@@ -64,6 +64,8 @@ export class Thread extends Component {
     setup() {
         super.setup();
         this.escape = escape;
+        this.refByMessageId = new Map();
```

Comment:
Maybe WeakMap, if we never need to do collection operations on it

Reply:
Good point, I'll adapt the PR :)

Reply:
This doesn't really need to be anything else than a basic JS object. It's just an id => value database.

Reply:
> This doesn't really need to be anything else than a basic JS object. It's just an id => value database.

JS engines assume objects keep relatively the same shape. If in practice their shape change frequently (= add props/remove props a lot), then this could result to more GC allocation than necessary, resulting in poor performance.

I don't think this is a big deal here, but just wanted to give some explanation on why `Map` is preferred over an object, although in terms of pure correctness they work the same here.